homeassistant/components/xiaomi_ble/config_flow.py

```diff
@@ -67,9 +158,21 @@ async def async_step_user(
         if user_input is not None:
             address = user_input[CONF_ADDRESS]
             await self.async_set_unique_id(address, raise_on_progress=False)
-            return self.async_create_entry(
-                title=self._discovered_devices[address], data={}
-            )
+            discovery = self._discovered_devices[address]
+
+            if discovery.device.encryption_scheme == EncryptionScheme.MIBEACON_LEGACY:
+                self._discovery_info = discovery.discovery_info
+                self._discovered_device = discovery.device
```

Comment:
`self._discovered_device` doesn't seem used after this step.

Reply:
Ah, it was originally used to set the config entry title but I cleaned up that to reduce duplication. I will remove this when I'm at a computer.

Reply:
Done in 2a4545048aef0ac46917e5975b493c271820ea04